json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java

```diff
@@ -1,15 +1,168 @@
+        Assert.assertEquals("a", Utils.concat("", "a", ""));
+        Assert.assertEquals("abc", Utils.concat("a", "b", "c"));
+    }
+
+    @Test
+    public void testEscape() {
+        Assert.assertNull(Utils.escape(null, true));
+
+        Assert.assertEquals("\\\\f\\'o\\\"o\\rb\\fa\\t\\nr\\bb\\/a",
+                Utils.escape("\\f\'o\"o\rb\fa\t\nr\bb/a", true));
+        Assert.assertEquals("\\uFFFF\\u0FFF\\u00FF\\u000F\\u0010",
+                Utils.escape("\uffff\u0fff\u00ff\u000f\u0010", true));
+    }
+
+    @Test
+    public void testUnescape() {
+        Assert.assertNull(Utils.unescape(null));
+
+        Assert.assertEquals("foo", Utils.unescape("foo"));
+        Assert.assertEquals("\\", Utils.unescape("\\"));
+        Assert.assertEquals("\\", Utils.unescape("\\\\"));
+        Assert.assertEquals("\'", Utils.unescape("\\\'"));
+        Assert.assertEquals("\"", Utils.unescape("\\\""));
+        Assert.assertEquals("\r", Utils.unescape("\\r"));
+        Assert.assertEquals("\f", Utils.unescape("\\f"));
+        Assert.assertEquals("\t", Utils.unescape("\\t"));
+        Assert.assertEquals("\n", Utils.unescape("\\n"));
+        Assert.assertEquals("\b", Utils.unescape("\\b"));
+        Assert.assertEquals("a", Utils.unescape("\\a"));
+        Assert.assertEquals("\uffff", Utils.unescape("\\uffff"));
+    }
+
+    @Test
+    public void testUnescapeThrow() {
+        thrown.expect(JsonPathException.class);
+        Utils.unescape("\\uuuuu");
+    }
 
-        String str =  "it\\\\";
+    @Test
+    public void testHex() {
+        Assert.assertEquals("61", Utils.hex('a'));
+        Assert.assertEquals("24", Utils.hex('$'));
+    }
+
+    @Test
+    public void testIsEmpty() {
+        Assert.assertTrue(Utils.isEmpty(null));
+        Assert.assertTrue(Utils.isEmpty(""));
 
+        Assert.assertFalse(Utils.isEmpty("foo"));
     }
 
+    @Test
+    public void testIndexOf() {
+        Assert.assertEquals(-1, Utils.indexOf("bar", "foo", 0));
+        Assert.assertEquals(-1, Utils.indexOf("bar", "a", 2));
+        Assert.assertEquals(1, Utils.indexOf("bar", "a", 0));
+        Assert.assertEquals(1, Utils.indexOf("bar", "a", 1));
+    }
+
+    @Test
+    public void testNotNull() {
+        Assert.assertEquals("", Utils.notNull("", "bar", "a", "b", "c"));
+        Assert.assertEquals("foo", Utils.notNull("foo", "bar", "a", "b", "c"));
+    }
+
+    @Test
+    public void testCloseQuietly() throws IllegalArgumentException {
+        thrown.expect(IllegalArgumentException.class);
+        Utils.notNull(null, "bar", "a", "b", "c");
+    }
+
+    @Test
+    public void testIsTrue() {
+        Utils.isTrue(true, "foo");
+    }
+
+    @Test
+    public void testIsTruethrows() {
+        thrown.expect(IllegalArgumentException.class);
+        Utils.isTrue(false, "foo");
+    }
+
+    @Test
+    public void testOnlyOneIsTrue() {
+        Utils.onlyOneIsTrue("foo", true, false);
+        Utils.onlyOneIsTrue("foo", false, true);
```

Comment:
is anything actually being tested here other than the methods are running?

Reply:
Yes.. its just checking if the method is run without throwing an exception

Reply:
Okay yeah i just had a look at it and it looks like that is all it does so there isn't really much you can do with that is there.

Reply:
test method removed